src/javaharness/java/arcs/api/ThingMapper.java

```diff
@@ -0,0 +1,69 @@
+package arcs.api;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class ThingMapper {
+    private String prefix;
+    private int nextIdentifier;
+    private final Map<String, Thing> idMap;
+    private final Map<Thing, String> reverseIdMap;
+
+    ThingMapper(String prefix) {
+        this.prefix = prefix;
+        this.nextIdentifier = 0;
+        idMap = new HashMap();
+        reverseIdMap = new HashMap();
+    }
+
+    private String newIdentifier(){
+        return this.prefix + (this.nextIdentifier++);
+    }
+
+    public String createMappingForThing(Thing thing, String requestedId) {
+        if (this.findMappingForThing(thing).isPresent()) {
```

Comment:
In Java, you don't need to prefix method calls with 'this', it's implicit unless there's shadowing.

Reply:
done